docs/source/user-guide/nlp/supported_tasks.rst

```diff
@@ -0,0 +1,100 @@
+.. _supported_tasks:
+
+===================================
+Working with Labels and Predictions
+===================================
+
+Some checks, mainly the ones related to model evaluation, require labels and model predictions in order to run.
+In the deepchecks nlp package, predictions are passed into the suite / check ``run`` method as pre-computed
+predictions only (passing a fitted model is currently not supported).
+
+
+.. _supported_task__types:
+
+Supported Task Types
+====================
+
+Deepchecks currently supports two NLP task types:
+
+* **Text Classification**: Text classification is any NLP task in which a whole body of text (ranging from a sentence
+  to a document) is assigned a class (in the binary/multiclass case) or a certain set of classes (in the multilateral
+  case). In both the binary, the multiclass and the multilabel case the class "belongs" / "classifies" the whole text
+  sample. Examples for such tasks are Sentiment Analysis, Topic Extraction, detecting harmful content and more.
+* **Token Classification**: Token Classification is any NLP task in which each word (or to be more accurate, token) in
+  the text sample is assigned a class of it's own. In many cases most tokens will belong to a "background" class,
+  allowing the model to focus on the interesting tokens. Examples for such tasks are Named Entity Recognition, Part
+  of Speach annotation (in which all tokens have a non-background class).
+
+.. _supported_labels__predictions_format:
+
+Supported Labels and Predictions Format
+=======================================
+
+While labels are passed while constructing the :class:`TextData <text_data.TextData>` object, predictions are passed
+separately to the ``run`` method of the check / suite. Labels and predictions must be in the format detailed in this
+section, according the task type.
+
+Label Formats:
+--------------
+
+* **Text Classification** Label - For text classification the accepted label format differs between multilabel and
+  single label cases. For single label data, the label should be passed as a sequence of labels, with one entry
+  per sample that can be either a string or an integer. For multilabel data, the label should be passed as a
+  sequence of sequences, with the sequence for each sample being a binary vector, representing the presence of
+  the i-th label in that sample.
+* **Token Classification** Label - For token classification the accepted label format is a sequence of sequences,
+  with the inner sequence containing tuples in the following format: (class_name, span_start, span_end).
+  span_start and span_end are the start and end character indices of the token within the text, as it was
+  passed to the raw_text argument. Each upper level sequence contains a sequence of tokens for each sample.
```

Comment:
```suggestion
  passed to the raw_text argument. Each outer sequence contains the sequence of tokens for each sample.
```

This sentence wasn't clear to me.. tried making it clearer (assuming that's what you meant)
Consider dropping completely this part, might make it clearer...